tests/helics/application_api/CMakeLists.txt

```diff
@@ -39,41 +38,30 @@ heat-transfer-tests.cpp
 ValueFederateExtendedTests.cpp
 )
 
-#  QUICK API TESTS
-
-add_executable(application-api-tests ${application_api_test_sources} ${application_api_test_headers})
+add_executable(application-api-tests ${application_api_test_sources} ${application_api_test_headers} ${key_test_files} ${extended_test_sources})
 target_link_libraries(application-api-tests helics-static helics_test_base)
 
-#  KEY TESTS
-add_executable(key-tests key-tests.cpp ${key_test_files} ${application_api_test_headers})
-target_link_libraries(key-tests helics-static helics_test_base)
-
-
-#  EXTENDED TESTS
-add_executable(extended-api-tests ${application_api_test_sources} ${extended_test_sources} ${application_api_test_headers})
-
-target_link_libraries(extended-api-tests helics-static helics_test_base)
-
-set_target_properties (application-api-tests extended-api-tests key-tests PROPERTIES FOLDER tests)
+set_target_properties (application-api-tests PROPERTIES FOLDER tests)
 
 target_compile_definitions(application-api-tests PRIVATE "-DHELICS_BROKER_LOCATION=\"${HELICS_BROKER_LOC}\"")
 target_compile_definitions(application-api-tests PRIVATE "-DHELICS_INSTALL_LOCATION=\"${CMAKE_INSTALL_PREFIX}\"")
-
 target_compile_definitions(application-api-tests PRIVATE "-DTEST_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}\"")
-
-target_compile_definitions(extended-api-tests PRIVATE "-DHELICS_BROKER_LOCATION=\"${HELICS_BROKER_LOC}\"")
-target_compile_definitions(extended-api-tests PRIVATE "-DHELICS_INSTALL_LOCATION=\"${CMAKE_INSTALL_PREFIX}\"")
-
-target_compile_definitions(extended-api-tests PRIVATE "-DTEST_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}\"")
 target_compile_definitions(application-api-tests PRIVATE "-DQUICK_TESTS_ONLY")
 
-target_compile_definitions(key-tests PRIVATE "-DKEY_TESTS")
-
-add_test(NAME key-tests COMMAND key-tests --log_level=message --report_level=short)
-add_test(NAME application-api-tests COMMAND application-api-tests --log_level=message --report_level=short)
-
-add_test(NAME extended-api-tests COMMAND extended-api-tests --log_level=message --report_level=short)
-
+add_test(NAME key-api-tests COMMAND application-api-tests --run_test=@key --log_level=message --report_level=short)
+add_test(NAME application-api-tests COMMAND application-api-tests --run_test=!@extended --log_level=message --report_level=short)
+add_test(NAME extended-api-tests COMMAND application-api-tests --log_level=message --report_level=short)
+set_property(TEST key-api-tests PROPERTY LABELS KeyApi)
```

Comment:
One of the other differences between the tests was the number of the test_case scenarios run in the DATA_TEST_CASES.  I think this is lost now with this setup.  I like the direction this is going but we might need to rethink how that is done since the extended test case is now very different from what it was.  see the testFixtures.hpp for the definitions.  I don't immediately know how we might deal with that other than to duplicate the data test cases and have one use the quick_test definitions, and another add the extended versions.

Reply:
with this you are defining QUICK_TESTS_ONLY for all the tests, which is different than before.

Reply:
The quick tests one is a problem, since that flag is also used to define an option that can comment out certain lines of a test. There isn’t any good way to control the subset of data driven test cases run, it seems to generate the cases at compile time — maybe something could be done with manual registration to pass in an argument listing the cores to use.